src/server/typingsInstaller/typingsInstaller.ts

```diff
@@ -332,6 +338,9 @@ namespace ts.server.typingsInstaller {
                     if (this.log.isEnabled()) {
                         this.log.writeLine(`Installed typings ${JSON.stringify(scopedTypings)}`);
                     }
+
+                    this.writePackageVersion(cachePath);

```

Comment:
why not write this in `processCacheLocation` when we discover that the two versions do not match (also having the side effect of wiping all the devdependencies already stored in package.json, and ensuring we do not see them again) instead of every time we attempt to install types?